tool/resources/org/antlr/v4/tool/templates/codegen/Java/Java.stg

```diff
@@ -915,11 +915,11 @@ public class <lexer.name> extends <superClass; null="Lexer"> {
 	public static final int
 		<rest(lexer.modes):{m | <m>=<i>}; separator=", ", wrap, anchor>;
 	<endif>
-	public static String[] channelNames = {
+	public static final String[] channelNames = {
```

Comment:
Hi, What is the advantage of make `channelNames` to `final`  while I can still modify internal data i.e : `channelNames[0] = "ma me mo"`.
Should be there a better solution ?

Reply:
Thanks for the comment.

The thing with the `final` modifier is that you cannot change the memory address of the variable. So `Class.channelNames = new String[] {}` will give you an error on compile time. If we want to make the content immutable.

If we want to make the collection itself immutable, we need to change to `List<String>`: https://docs.oracle.com/en/java/javase/12/core/creating-immutable-lists-sets-and-maps.html
I'm fine with that as well, but then we're changing the public API. In other words, if some people are using the `channelNames` and `modeNames` in their application, then they need to update their application code as well. We can do this, but then we should wait for a major release.

Any opinion on this @parrt ?